packages/core/react/src/hooks.ts

```diff
@@ -0,0 +1,898 @@
+import type {
+    AccountBalancePair,
+    AccountInfo,
+    Commitment,
+    ConfirmedSignatureInfo,
+    EpochInfo,
+    EpochSchedule,
+    GetBlockProductionConfig,
+    GetLargestAccountsConfig,
+    GetParsedProgramAccountsConfig,
+    GetProgramAccountsConfig,
+    GetProgramAccountsResponse,
+    GetVersionedTransactionConfig,
+    InflationGovernor,
+    InflationRate,
+    InflationReward,
+    ParsedAccountData,
+    RpcResponseAndContext,
+    StakeActivationData,
+    TokenAccountBalancePair,
+    TokenAmount,
+    TransactionResponse,
+    Version,
+    VersionedTransactionResponse,
+    VoteAccountStatus,
+} from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import { useEffect, useState } from 'react';
+import { useConnection } from './useConnection.js';
+
+const TOKEN_PROGRAM_ID = new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA');
+const ASSOCIATED_TOKEN_PROGRAM_ID = new PublicKey('ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL');
+
+export type HookOptions = {
+    refetchInterval?: number | null;
+};
+
+export type HookReturnType<T> = {
+    data: T | null;
+    refetch: () => void;
+    isLoading: boolean;
+    error: any;
+};
+
+export function useBalance(
+    publicKey: PublicKey | null,
+    { refetchInterval = null }: HookOptions = {}
+): HookReturnType<number> {
+    const [balance, setBalance] = useState<number | null>(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<any>(null);
+    const { connection } = useConnection();
+
+    const fetchBalance = async () => {
+        if (!publicKey) return;
+        try {
+            setError(null);
+            setIsLoading(true);
+            const res = await connection.getBalance(publicKey);
+            setBalance(res);
```

Comment:
We should add a way to abort the call when the publicKey changes so the old calls get cleaned up.

Case:

- A wallet is connected
- The fetch function is triggered, this is called 1
- The wallet switches to another account
- The fetch function is triggered again, this is called 2
- Called 1 finishes and populates the states with old data

This is one of the things React Query prevents, it's a very common use case that may affect to any other hook as well

Reply:
mhhh does this not mean we need some kind of abort controller inside the rpc calls?

Reply:
Maybe we can add something like this
https://react.dev/learn/removing-effect-dependencies#is-your-effect-doing-several-unrelated-things